pkg/autoscaler/autoscaler.go

```diff
@@ -148,31 +103,22 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (int32, bool) {
 	// Use 1 if there are zero current pods.
 	readyPodsCount := math.Max(1, float64(originalReadyPodsCount))
 
-	// Remove outdated data.
-	a.buckets.RemoveOlderThan(now.Add(-spec.MetricSpec.StableWindow))
-	if a.buckets.IsEmpty() {
-		logger.Debug("No data to scale on.")
+	metricKey := NewMetricKey(a.namespace, a.revision)
+	observedStableConcurrency, observedPanicConcurrency, err := a.metricClient.StableAndPanicConcurrency(metricKey)
+	if err != nil {
```

Comment:
Looks like the behavior will change. Before: do not scale if the bucket is empty. After: scale to 0 if the bucket is empty. If there're some issue with metrics pipeline and no data are sent to collector, the pods will be scaled to 0. I don't think this is desired behavior.

Reply:
Hmm good call. I guess we need to propagate an error from the metrics collector here.

Reply:
Done!